llvm/lib/Target/X86/X86InsertWait.cpp

```diff
@@ -115,7 +115,8 @@ bool WaitInsert::runOnMachineFunction(MachineFunction &MF) {
       // If the following instruction is an X87 instruction and isn't an X87
       // non-waiting control instruction, we can omit insert wait instruction.
       MachineBasicBlock::iterator AfterMI = std::next(MI);
-      if (AfterMI != MBB.end() && X86::isX87Instruction(*AfterMI) &&
+      if (AfterMI != MBB.end() && !AfterMI->isCall() &&
+          X86::isX87Instruction(*AfterMI) &&
           !isX87NonWaitingControlInstruction(*AfterMI))
```

Comment:
Maybe move `isCall` check into `isX87Instruction`?

Reply:
What about the MBB.end() then since this would make sense here as well?

Reply:
`MBB.end()` is different. It is not a MI.

Reply:
> Maybe move `isCall` check into `isX87Instruction`?

Why? It makes it unclear where the isCall check is used.

Reply:
Because the code in `isX87Instruction` just checked for X87 registers before. Call defs X87 register but should not be classified as X87 instruction.